### PR TITLE
Ethans gconf comments

### PR DESCRIPTION
**Pull request by ethanfrey**
Code speaks a thousand docs.

I liked this approach. But made some changes. Please look at the diff.

1. removed magic. pass in the package name/key rather than auto-querying it
2. enforce validation on parsing
3. abstracted out some basic InitConfig logic to make it easy to integrate in each package
4. Ensure tests in gconf and x/cash pass
5. changed genesis format to 
```json
"conf": {
  "cash": {..},
...
```

Please review the code and let me know what you think. If you are good, we can clean up and use this approach

### CODE REVIEW

File: gconf/gconf.go [hunk 1]
```diff
@@ -46,17 +54,27 @@ type Unmarshaler interface {
...
+// Returns an error if anything goes wrong
+func InitConfig(db Store, opts weave.Options, pkg string, conf Configuration) error {
+	var confOptions weave.Options
+	if err := opts.ReadOptions("conf", &confOptions); err != nil {
+		return errors.Wrap(err, "read conf")
 	}
-	if t.Kind() != reflect.Struct {
-		return ""
+	if confOptions[pkg] == nil {
+		return errors.Wrapf(errors.ErrInvalidInput, "No configuration for %s", pkg)
```
Comment (husio) [inline gconf/gconf.go]: Please don't capitalize error strings https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: gconf/gconf.go [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
+// Store is a subset of weave.KVStore
 type Store interface {
 	Get([]byte) []byte
 	Set([]byte, []byte)
 }
 
-func Save(db Store, src Marshaler) error {
-	key := []byte("configuration:" + pkgPath(src))
+// Save will validate code
```
Comment (husio) [inline gconf/gconf.go]: Please write useful documentation :wink: :policeman:

File: x/cash/codec.proto [hunk 3]
```diff
@@ -35,6 +35,7 @@ message FeeInfo {
 
 
 message Configuration {
+  // TODO: add schema uint32 here
```
Comment (husio) [inline x/cash/codec.proto]: This should land in https://github.com/iov-one/weave/issues/532 after merging. I think it's better to implement it before merging.

**Comment by ethanfrey** [comment]
Yeah, I was a bit sloppy on the details. If Alex likes the idea as well, we can flesh it out. I would also add validation rules and see how to update schema/config before merging to master. 

But if we agree this is the direction we go, we can start working on the details. In office today or remote

**Comment by ethanfrey** [comment]
@husio This branch should be good to go.

I added the generic handler with some basic tests and also fixed all tests in cmd/* and examples/mycoin.  Please run `make test` locally

If you like this, please merge it into your PR, rebase that PR on newest master, and then add any more changes you want to do to consider this finished. I am sure there is plenty left to improve, but the basic flow should work and captures all difficult logic in gconf.

**Comment by ethanfrey** [comment]
Oh, and this will need a nicely updated CHANGELOG before merging to master :wink:

**Pull request merged**